parser.go

```diff
@@ -57,6 +57,30 @@ func listInventory() {
 	}
 }
 
+// moveToRoom takes a requested exit and moves the player there if the exit exists
+func moveToRoom(exit string) {
+	for _, e := range curRoom.Exits {
+		if e == exit { // check that requested exit is valid
+			for i, j := range rooms { // find the room it leads to
+				if j.Name == exit {
+					curRoom = rooms[i] // if found, the exit is the new current room
+
+					if curRoom.Visited == false { // have we been here before?
+						curRoom.Visited = true
+						rooms[i] = curRoom
```

Comment:
What does this line do?

Reply:
If we have not visited the current room before, then mark it as visited. 
After updating the current room's Visited field, replace that room in the slice rooms with the updated version.